### PR TITLE
Use existing selfDM report 

### PR DESCRIPTION
**Pull request by jnowakow**
### Explanation of Change

SelfDM reports will be created during account creation and will have notification preference equal to `hidden` by default. Optimistically creating selfDM reports during onboarding is causing bugs with new behaviour so this PR removes that logic

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/556957
PROPOSAL: n/a

### Tests

1. Create new account
2. Fill onboarding using any path
3. Verify that it's possible to find selfDM
4. Verify that after any action selfDM notification preference is updated to `muted`. 

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Create expense on workspace chat
2. Go offline
3. Go to Expense
4. Click on report field and "Remove from report"
5. Verify that self DM is created
6. Go online
7. Verify that there's only one self DM chat and nothing flickers 

### QA Steps

// TODO: These must be filled out, or the issue title must include "[No QA]."

1. Create new account
2. Fill onboarding using any path
3. Verify that it's possible to find selfDM
4. Verify that after any action selfDM notification preference is updated to `muted`. 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified there are no new alerts related to the `canBeMissing` param for `useOnyx`
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L80)
      - [x] If any non-english text was added/modified, I used [JaimeGPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) to get English > Spanish translation. I then posted it in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L116-L123)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If new assets were added or existing ones were modified, I verified that:
    - [x] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [x] The assets load correctly across all supported platforms.
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/c6a08ae6-413a-408e-b41a-f120fe6e1d36

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/b175a9f1-ffd2-458f-afa6-c16fca3744ab

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/1c6c5790-c3d3-42c2-a8fe-169d805b4128

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/45a5a326-3319-4d8a-b6df-05e86f1b215d

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/libs/actions/Report.ts [hunk 1]
```diff
@@ -4957,72 +4956,7 @@ function deleteAppReport(reportID: string | undefined) {
 
     const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
 
-    let selfDMReportID = findSelfDMReportID();
```
Comment (mountiny) [inline src/libs/actions/Report.ts:111]: If the selfDM report is created as hidden initially, the OpenApp will not return it, right? Then this would fail still? 

So if you:
1. Create a new account and selfDM is created as hidden
2. OpenApp does not return it after a sign-out and sign in
3. Create an expense on the workspace
4. Delete report
5. The expenses should move to the selfDM

**Review by mountiny — COMMENTED** [review]
@jnowakow should this be on hold for the web changes?

File: src/libs/actions/Report.ts (hunk 1 above)
Comment (jnowakow) [reply src/libs/actions/Report.ts:111]: `OpenApp` doesn't return selfDM report but it does return [its ID](https://github.com/Expensify/Auth/blob/fa0917c8aec85511085ab012fa5e82e05b8a8500/auth/command/OpenApp.cpp#L263-L266)

**Comment by jnowakow** [comment]
> @jnowakow should this be on hold for the web changes?

Right, rollout plan is [here](https://github.com/Expensify/Web-Expensify/pull/49121/files#r2517913366). I'm putting it on hold again

File: src/libs/actions/Report.ts (hunk 1 above)
Comment (mountiny) [reply src/libs/actions/Report.ts:111]: So then when we move the user to the report optimistically, how will it look like if they are offline but they dont have the report data, created report action and metadata?

Comment (jnowakow) [reply src/libs/actions/Report.ts:111]: `SelfDM` report is created in an optimistic way with correct `reportID`. Report metadata and report action are also created in optimistic way. But since we don't know reportAction ID, the action is removed when requested is finished successfully and correct one from backend is inserted into Onyx.

**Review by trjExpensify — DISMISSED** [review]
We agreed to make this change in Slack, all good. 👍

File: src/libs/ReportUtils.ts [hunk 2]
```diff
@@ -11522,73 +11526,6 @@ function prepareOnboardingOnyxData(
...
     }
 
-    let selfDMParameters: SelfDMParameters = {};
-    if (engagementChoice === CONST.ONBOARDING_CHOICES.PERSONAL_SPEND || engagementChoice === CONST.ONBOARDING_CHOICES.TRACK_WORKSPACE) {
-        const selfDMReportID = findSelfDMReportID();
-        let selfDMReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${selfDMReportID}`];
-        let createdAction: ReportAction;
-        if (!selfDMReport) {
-            const currentTime = DateUtils.getDBTime();
-            selfDMReport = buildOptimisticSelfDMReport(currentTime);
```
Comment (jnowakow) [inline src/libs/ReportUtils.ts:11698]: At this time users will have selfDM created although it won't be in their onyx. Probably instead of creating it optimistically it would be better to return it in `CompleteGuidedSetup`

File: src/libs/ReportUtils.ts [hunk 3]
```diff
@@ -1201,6 +1201,17 @@ Onyx.connect({
...
+Onyx.connectWithoutView({
+    key: ONYXKEYS.SELF_DM_REPORT_ID,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+        selfDMReportID = value;
+    },
+});
+
```
Comment (jnowakow) [inline src/libs/ReportUtils.ts:1220]: This key is pushed from OpenApp since around month. @mountiny is it safe to assume that it will be present on all accounts?

Comment (mountiny) [reply src/libs/ReportUtils.ts:1220]: Yes I think its ok, but also I believe the plan is still to fallback to the optimistic report creation

Comment (jnowakow) [reply src/libs/ReportUtils.ts:1220]: Yes, but previously to find selfDM we were iterating through reports and I'm wondering if it's possible that there will be selfDM in reports collection but for some reason there won't be `selfDMReportID` key in Onyx

Comment (mountiny) [reply src/libs/ReportUtils.ts:1220]: Got it, I think we can just take this key at this point

Comment (mountiny) [reply src/libs/ReportUtils.ts:1220]: Actually @shubham1206agra Do you have some plan to update this? Maybe we could leave this out of scope of this PR (removal of `findSelfDMReportID` and just use it as a fallback if the SELF_DM_REPORT_ID is not defined)

Comment (shubham1206agra) [reply src/libs/ReportUtils.ts:1220]: @mountiny You can leave it for this PR. I will migrate it slowly as there are way too many references.

**Comment by eh2077** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified that the composer does not automatically focus or open the keyboard on mobile unless explicitly intended. This includes checking that returning the app from the background does not unexpectedly open the keyboard.
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: HybridApp
    - [ ] Android: mWeb Chrome
    - [ ] iOS: HybridApp
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified there are no new alerts related to the `canBeMissing` param for `useOnyx`
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L80)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L116-L123)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: HybridApp</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: HybridApp</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/libs/actions/Report.ts
Comment (eh2077) [inline src/libs/actions/Report.ts:1]: There's a user case of `findSelfDMReportID` left to be migrated.

File: src/libs/actions/Transaction.ts [hunk 4]
```diff
@@ -774,9 +780,8 @@ function changeTransactionsReport(
                 onyxMethod: Onyx.METHOD.MERGE,
                 key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${selfDMReport.reportID}`,
                 value: {
-                    [selfDMCreatedReportAction.reportActionID]: {
-                        pendingAction: null,
-                    },
+                    // created self DM action has different ID so we have to clean it
+                    [selfDMCreatedReportAction.reportActionID]: null,
```
Comment (eh2077) [inline src/libs/actions/Transaction.ts]: Hmm, this change doesn't make sense to me. I think we usually only clear `pendingFields` and `pendingAction` for success request.

Or do you mean clearing `ONYXKEYS.SELF_DM_REPORT_ID` here?

File: src/libs/ReportUtils.ts (hunk 3 above)
Comment (mountiny) [reply src/libs/ReportUtils.ts:1220]: @jnowakow would you be able to update the PR? I think we should not include it in the scope here?

Comment (jnowakow) [reply src/libs/ReportUtils.ts:1220]: Why we shouldn't use it here? Without this we won't be able to find selfDM reportID for newly created accounts 🤔

Comment (shubham1206agra) [reply src/libs/ReportUtils.ts:1220]: @mountiny Can you create an issue for this Onyx.connect refactor and assign it to me?

Comment (mountiny) [reply src/libs/ReportUtils.ts:1220]: @jnowakow What I mean is keep `findSelfDMReportID` as is (looping over the reports) and keep this onyx connect and default to the selfReportID from the onyx key if there is no selfDM found in reports.

This si the most defensive we can take here, I dont want to cause regressions because we removed the logic that searches for the selfDM report

Comment (mountiny) [reply src/libs/ReportUtils.ts:1220]: @shubham1206agra let me know if DM with the format of the issue I can create it for you but now not sure if I understand exactly what you asking me for

Comment (jnowakow) [reply src/libs/ReportUtils.ts:1220]: Okey, fallback to `findSelfDMReportID` makes sense

File: src/libs/actions/Transaction.ts (hunk 4 above)
Comment (jnowakow) [reply src/libs/actions/Transaction.ts]: Good catch!

File: src/libs/ReportUtils.ts (hunk 3 above)
Comment (jnowakow) [reply src/libs/ReportUtils.ts:1220]: @mountiny I've updated the PR to fallback to `findSelfDMReportID` if there's nothing under `SELF_DM_REPORT_ID` key in Onyx :D

File: src/libs/ReportUtils.ts [hunk 5]
```diff
@@ -1848,15 +1859,31 @@ function isConciergeChatReport(report: OnyxInputOrEntry<Report>): boolean {
...
     if (!allReports) {
         return;
     }
 
     const selfDMReport = Object.values(allReports).find((report) => isSelfDM(report) && !isThread(report));
+
+    // If we found the self DM report, set the self DM report ID in Onyx for quicker access
+    if (selfDMReport) {
+        Onyx.set(ONYXKEYS.SELF_DM_REPORT_ID, selfDMReport.reportID);
+    }
```
Comment (mountiny) [inline src/libs/ReportUtils.ts]: Let's not do this now, I think this is solving a different problem than the one this pr was aiming for.

Instead, lets exit early from this method if the key exist with reportID, if no, lets continue using the loop over the reports -other prs will take care of this

File: src/libs/ReportUtils.ts [hunk 6]
```diff
@@ -1848,15 +1859,31 @@ function isConciergeChatReport(report: OnyxInputOrEntry<Report>): boolean {
...
 }
 
+/**
+ * Returns the self DM report ID
+ */
+function getSelfDMReportID(): string | undefined {
+    // eslint-disable-next-line @typescript-eslint/no-deprecated
+    return selfDMReportID ?? findSelfDMReportID();
+}
+
```
Comment (mountiny) [inline src/libs/ReportUtils.ts]: We dont need this method if we have the previous one now? Lets just keep using `findSelfDMReportID` and dont mark it as deprecated, again this will be done in other PRs

**Review by mountiny — CHANGES_REQUESTED** [review]

**Comment by mountiny** [comment]
@eh2077 let us know how the testing continues 🙌

**Comment by eh2077** [comment]
Bug 1: click avatar of selfDM opens profile page of Concierge
Bug 2: Title of selfDM chat isn't in format `${user_display_name} (you)` but `${user_display_name}, Expensify`


https://github.com/user-attachments/assets/3b239412-709b-455a-860c-de530709a61c

**Comment by eh2077** [comment]
Bug 3: Title of selfDM chat isn't consistent, sometimes it's in format `${user_display_name} (you)`


https://github.com/user-attachments/assets/eef682d6-6fdd-4afe-ba71-7963af39184c

**Comment by jnowakow** [comment]
I feel like all four bugs are the same issue and they are not caused by this PR since it already exists in staging. It seems that you get different selfDM report name depending if you have tracked expense or started chat.
When I try to track expense I'm getting "<you>, Expensify" chat name: 

https://github.com/user-attachments/assets/83d6f6d3-e9c5-490c-821d-f24cb7d1803c

When I choose "start chat" option and find my selfDM then I'm getting "<username> (you)" chat name:

https://github.com/user-attachments/assets/a36fc5ee-3d3c-437d-8c79-785336137ae2

**Comment by trjExpensify** [comment]
@MonilBhavsar if the user selected the Track intent, should the selfDM be `hidden`? I don't recall that.. 🤔

**Review by eh2077 — APPROVED** [review]
Since all issues happen on staging too, approving to move it forward!

**Comment by MonilBhavsar** [comment]
> I think we can do it in separate PR as it would require also some backend work and is out of scope of the [original issue](https://github.com/Expensify/Expensify/issues/556957). What do you think? :D

Agree, this is a separate bug and happy to handle it myself separately.

**Comment by jnowakow** [comment]
> > I think we can do it in separate PR as it would require also some backend work and is out of scope of the [original issue](https://github.com/Expensify/Expensify/issues/556957). What do you think? :D
> 
> Agree, this is a separate bug and happy to handle it myself separately.

So I think we're good to proceed with this PR :D Can you do the review @MonilBhavsar? 🙏

File: src/libs/actions/IOU.ts [hunk 7]
```diff
@@ -4002,9 +4002,10 @@ function getTrackExpenseInformation(params: GetTrackExpenseInformationParams): T
     // STEP 1: Get existing chat report
     let chatReport = !isEmptyObject(parentChatReport) && parentChatReport?.reportID ? parentChatReport : null;
 
+    const selfDMReportID = findSelfDMReportID();
     // If no chat report is passed, defaults to the self-DM report
     if (!chatReport) {
-        chatReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${findSelfDMReportID()}`] ?? null;
+        chatReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${selfDMReportID}`] ?? null;
```
Comment (MonilBhavsar) [inline src/libs/actions/IOU.ts]: We can revert this change?

File: src/libs/actions/Report.ts [hunk 8]
```diff
@@ -4845,6 +4846,11 @@ function deleteAppReport(reportID: string | undefined) {
                     [createdAction.reportActionID]: createdAction,
                 },
             },
+            {
+                onyxMethod: Onyx.METHOD.SET,
+                key: ONYXKEYS.SELF_DM_REPORT_ID,
+                value: selfDMReport.reportID,
+            },
```
Comment (MonilBhavsar) [inline src/libs/actions/Report.ts]: Do we need to set this here? Why?

**Review by MonilBhavsar — COMMENTED** [review]
Could you also please update PR title

File: src/libs/actions/Report.ts (hunk 8 above)
Comment (jnowakow) [reply src/libs/actions/Report.ts]: It's not strictly needed. I'll revert this and allow to handle this new onyx key to be handled in separate PR

File: src/libs/actions/IOU.ts (hunk 7 above)
Comment (jnowakow) [reply src/libs/actions/IOU.ts]: Reverted

File: tests/actions/ReportTest.ts [hunk 9]
```diff
@@ -2058,7 +2059,6 @@ describe('actions/Report', () => {
                     },
                 });
             });
-            // The length is 3 to include the CREATED action
```
Comment (MonilBhavsar) [inline tests/actions/ReportTest.ts:2060]: Can we please restore the comment too?

Comment (jnowakow) [reply tests/actions/ReportTest.ts:2060]: Done :)

Comment (MonilBhavsar) [reply tests/actions/ReportTest.ts:2060]: Thanks!

File: src/libs/actions/Report.ts (hunk 8 above)
Comment (MonilBhavsar) [reply src/libs/actions/Report.ts]: Sounds good 👍 Thanks!

**Review by MonilBhavsar — DISMISSED** [review]
Thanks!

**Comment by MonilBhavsar** [comment]
Looks like it worked! @mountiny if you can take a quick look and merge since you requested some changes

File: src/libs/actions/Transaction.ts [hunk 10]
```diff
@@ -719,14 +719,15 @@ function changeTransactionsReport(
     const failureData: OnyxUpdate[] = [];
     const successData: OnyxUpdate[] = [];
 
-    const existingSelfDMReportID = findSelfDMReportID();
-    let selfDMReport: Report | undefined;
+    let selfDMReportID = findSelfDMReportID();
+    let selfDMReport: Report | undefined = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${selfDMReportID}`];
```
Comment (mountiny) [inline src/libs/actions/Transaction.ts:724]: Not sure if this method is pure, but we should pass the selfDM report in here to make it pure

File: src/libs/ReportUtils.ts [hunk 11]
```diff
@@ -1857,7 +1868,12 @@ function findSelfDMReportID(): string | undefined {
         return;
     }
 
+    if (selfDMReportID) {
+        return selfDMReportID;
+    }
+
     const selfDMReport = Object.values(allReports).find((report) => isSelfDM(report) && !isThread(report));
+
     return selfDMReport?.reportID;
```
Comment (mountiny) [inline src/libs/ReportUtils.ts:1877]: Would be great to cover this logic in Unit tests if there are none yet, maybe lets look into that in a follow up

**Review by mountiny — DISMISSED** [review]
Thanks!

File: src/libs/ReportUtils.ts (hunk 11 above)
Comment (jnowakow) [reply src/libs/ReportUtils.ts:1877]: I'll create follow-up with this change then :)

File: src/libs/actions/Transaction.ts (hunk 10 above)
Comment (jnowakow) [reply src/libs/actions/Transaction.ts:724]: @mountiny updated :)

**Comment by MonilBhavsar** [comment]
tests are failing @jnowakow

File: src/libs/actions/Transaction.ts (hunk 10 above)
Comment (mountiny) [reply src/libs/actions/Transaction.ts:724]: I think you would need to pass the report, not just the id to make it pure, but lets not do it here

**Review by mountiny — APPROVED** [review]

**Review by mountiny — APPROVED** [review]
Perfect, thank you! all yours @MonilBhavsar
